contrib/uriparser/example/main.cc

```diff
@@ -0,0 +1,116 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "absl/flags/parse.h"
+#include "contrib/uriparser/sandboxed.h"
+#include "contrib/uriparser/utils/utils_uriparser.h"
+
+void print(const char* name, absl::StatusOr<std::string> r) {
+  if (!r.ok()) {
+    std::cerr << "Unable to fetch " << name << "\n";
+    std::cerr << r.status() << "\n";
+    return;
+  }
+
+  if (r.value().empty()) {
+    return;
+  }
+
+  std::cout << name << ": " << *r << "\n";
+}
+
+int main(int argc, char* argv[]) {
+  std::string prog_name(argv[0]);
+  google::InitGoogleLogging(argv[0]);
+  std::vector<char*> args = absl::ParseCommandLine(argc, argv);
+
+  if (args.size() < 2) {
+    std::cerr << "Usage:\n  " << prog_name << " URI ...\n";
+    return EXIT_FAILURE;
+  }
+
+  UriparserSapiSandbox sandbox;
+  if (!sandbox.Init().ok()) {
+    std::cerr << "Unable to start sandbox\n";
+    return EXIT_FAILURE;
+  }
+
+  int retval = EXIT_SUCCESS;
+  for (int i = 1; i < args.size(); i++) {
+    Uriparser uri(&sandbox, args[i]);
+    if (!uri.GetStatus().ok()) {
+      std::cerr << "Unable to parse: " << args[i] << "\n";
+      std::cerr << uri.GetStatus() << "\n";
+      retval = EXIT_FAILURE;
+      continue;
+    }
+
+    print("scheme", uri.GetScheme());
+    print("user info", uri.GetUserInfo());
+    print("host", uri.GetHostText());
+    print("host IP", uri.GetHostIP());
+    print("port", uri.GetPortText());
+    print("query", uri.GetQuery());
+    print("fragment", uri.GetFragment());
+
+    absl::StatusOr<std::vector<std::string>> path = uri.GetPath();
+    if (!path.ok()) {
+      std::cerr << "Unable to get path.\n";
+      std::cerr << path.status() << "\n";
+      retval = EXIT_FAILURE;
+      continue;
+    }
+    if (path.value().size() > 0) {
```

Comment:
Optional: when you know that a StatusOr can only contain a value, you can use its `->` operator:
```c++
if (path->size() > 0) {...
```

In this case, instead of checking for `size()` I'd also prefer to check against `empty()` for readability.

Reply:
Fixed.